src/components/ListItem.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useListContext } from '../context/ListContext';
+
+import { useState } from 'react';
+import styles from '../styles.css';
+
+export default function ListItem({ item }) {
+  const { ACTIONS, dispatch } = useListContext();
+  const [editing, setEditing] = useState(false);
+  const [name, setName] = useState(item.name);
+
+  const handleToggle = () => {
+    dispatch({ type: ACTIONS.TOGGLE_ITEM, payload: { id: item.id } });
+  };
+
+  const handleDelete = () => {
+    dispatch({ type: ACTIONS.DELETE_ITEM, payload: { id: item.id } });
+  };
+
+  const handleEdit = (e) => {
+    e.preventDefault();
+    dispatch({ type: ACTIONS.EDIT_ITEM, payload: { id: item.id, name: name } });
+    setEditing(false);
+  };
+
+  return (
+    <span>
+      {editing ? (
```

Comment:
Another option here would be to save this jsx into a variable and call it within the return().